pkg/controller/statusmanager/status_manager.go

```diff
@@ -85,7 +86,7 @@ func (status *StatusManager) Set(reachedAvailableLevel bool, conditions ...condi
 // set updates the NetworkAddonsConfig.Status with the provided conditions
 func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...conditionsv1.Condition) error {
```

Comment:
We are doing two things here, creating the spected status and storing it at the versioned version, let's split that so we do the versined part ("storing the status") out of here.

Reply:
you mean you want a function like `getNetworkAddonsConfigV1()`?

Reply:
will do it as part of the following PR using the new `NetworkAddonsStorageVersion` interface